src/branch.c

```diff
@@ -468,7 +468,7 @@ int git_branch_upstream_name(
 	return error;
 }
 
-int git_branch_upstream_remote(git_buf *buf, git_repository *repo, const char *refname)
+static int git_branch_upstream_with_format(git_buf *buf, git_repository *repo, const char *refname, const char *format)
```

Comment:
I like what you've done here.  But if we're making this more generic, we'll want to take another argument that indicates what we're fetching, and use it in the error message in this function (line 487).

Reply:
Could you suggest this argument?

I think that it is better to add enum for formats

```c
typedef enum {
  GIT_BRANCH_UPSTREAM_FORMAT_REMOTE = 1,
  GIT_BRANCH_UPSTREAM_FORMAT_MERGE = 2
} git_branch_upstream_format;

static const char* git_branch_upstream_format_for_id(git_branch_upstream_format id) {
  switch (id) {
  case GIT_BRANCH_UPSTREAM_FORMAT_REMOTE: return "branch.%s.remote";
  case GIT_BRANCH_UPSTREAM_FORMAT_MERGE: return "branch.%s.merge";
  default: return ""; // OK?
  };
}
```

And determine error message by this enum argument.